src/VirtoCommerce.MarketingModule.Data/Handlers/CouponUsageRecordHandler.cs

```diff
@@ -78,6 +82,24 @@ private async Task RecordUsages(string objectId, IEnumerable<PromotionUsage> old
             }
         }
 
+        private async Task ValidateUsagesAndThrow(IEnumerable<PromotionUsage> usages)
+        {
+            var invalidUsage = usages.FirstOrDefault(x => x.PromotionId.IsNullOrEmpty());
```

Comment:
We don't need to create the variable `invalidUsage`. We can use `Any()` in conditions.
``` csharp
// Promotion is missing
if (usages.Any(x => x.PromotionId.IsNullOrEmpty()))
{
    throw new ValidationException($"Invalid PromotionUsage: CouponCode '{invalidUsage.CouponCode}' for objectId '{invalidUsage.ObjectId}' of type {invalidUsage.ObjectType}. PromotionId is missing!");
}

var promotionIds = usages.Select(x => x.PromotionId).ToArray();
var promotions = await _promotionService.GetPromotionsByIdsAsync(promotionIds);

// Promotion not found
if (usages.Any(x => !promotions.Any(p => p.Id == x.PromotionId)))
{
    throw new ValidationException($"Invalid PromotionUsage: CouponCode '{invalidUsage.CouponCode}' for objectId '{invalidUsage.ObjectId}' of type {invalidUsage.ObjectType}. Promotion with Id '{invalidUsage.PromotionId}' not found!");
}
```

Reply:
I know Any(), but the invalidUsage here is essential to form the resulting exception, not just for checking the condition.